site/content/blog/making-the-most-out-of-clis.md

```diff
@@ -0,0 +1,93 @@
+---
+title: "Making the most out of CLIs"
+slug: making-the-most-out-of-clis
+date: 2022-03-11
+author: Soumik Majumder
+excerpt: "Find out how different Carvel CLIs interacting with resources on the cluster and kubectl fit together like pieces of a jigsaw puzzle"
+image: /img/logo.svg
+tags: ['Soumik Majumder', 'carvel', 'kctrl']
+---
+
+Carvel is glad to have `kapp` and the (brand new!) `kctrl` as a part of our arsenal. In this blog, we will see how these powerful CLIs, along with our good old friend `kubectl`, fit into our day-to-day workflows.
+
+We often see our users leverage all three of them in their workflows. Let's look at what each one of them is super good at!
+
+## `kubectl`
+`kubectl` is a CLI tool built by the Kubernetes team. It let's users interact with the cluster and resources on it.
+
+### Viewing resources current state on the cluster
+`kubectl` lets you fetch and view the current state of a resource on the cluster in it's entirety. This is quite useful while exploring resources on a cluster and debugging faulty resources.
+
+### Managing your `kubeconfig` and contexts
+The set of commands under `kubectl config` allow us to manage and modify the configurations we use while connecting to our clusters. These configurations include values which indicate which cluster the tool interacts with, how we authenticate with the cluster and default namespace used.
+This is used as the default configuration for connecting to a cluster by other tools on the host - more often than not. And `kubectl` allows us to seamlessly modify and switch between multiple "contexts" like these.
+
+### Interacting with nodes
+`kubectl` has a set of powerful commands that let us cordon and uncordon nodes (making them available or not available for scheduling). These commands can also help users drain resources from a node so that it can be cordoned safely. Nodes can be "tainted" with certain key-values, which allow only pods that tolerate those "taints" to be scheduled on them.
+Learn more about these commands [here](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#-strong-cluster-management-strong-).
+
+### Dive into your containers and applications
+`kubectl` allows us to attach to processes containers running on pods and execute commands inside them easily. Users can observe the resource consumption on a pod as well.
+// might remove
+Proxying and port-forwarding options let a pod or the K8s API server serve HTTP content to localhost. Letting users debug their applications easily
+
+## `kapp`
+`kapp` is a CLI team that helps users manage their workloads effectively on the cluster. This involves defining interactions between resources declaratively, deploying resources reliably and updating them safely and confidently.
```

Comment:
Did you mean `built by Carvel team`?
```suggestion
`kapp` is a CLI built by the Carvel team that helps users manage their workloads effectively on the cluster. This involves defining interactions between resources declaratively, deploying resources reliably and updating them safely and confidently.
```

Reply:
It was a typo but I believe I can add that

Reply:
Removing it all together since we mention in the beginning that kctrl and kapp are built by us